sycl/source/detail/scheduler/commands.hpp

```diff
@@ -107,9 +107,11 @@ class Command {
 
   Command(CommandType Type, QueueImplPtr Queue);
 
-  void addDep(DepDesc NewDep);
+  /// \return an optional connection cmd to enqueue
+  [[nodiscard]] Command *addDep(DepDesc NewDep);
```

Comment:
Just to pay attention, can we really use [[nodiscard]] attribute now, because it is since C++17 ?

Reply:
Good point! I remember @romanovvlad asked to stop using C++17 features in common code that may need to stay compatible with C++14

Reply:
Was this request for headers i.e. for the code which is available for user?
`commands.hpp` isn't available for user.

@romanovvlad , could you, please, clarify here?

Reply:
Using C++17 features in `source` dir is fine.